### PR TITLE
Add support for answer action

### PR DESCRIPTION
**Pull request by MebinAbraham**
### PR Context
Added support for answer actions feature introduced in https://github.com/ONSdigital/eq-survey-runner/pull/2301

### How to review
Test with https://github.com/ONSdigital/eq-survey-runner/pull/2301

### CODE REVIEW

**Comment by MebinAbraham** [comment]
[Ability to route to a List Collector's add block.](https://trello.com/c/Yrl8kLIk/3221-ability-to-route-to-a-list-collectors-add-block)

File: app/validation/validator.py [hunk 1]
```diff
@@ -229,6 +232,26 @@ def _validate_questions(self, block_or_variant, numeric_answer_ranges):
 
         return errors
 
+    def _validate_answer_actions(self, answer):
+        errors = []
+        answer_options = answer.get('options', {})
+        for option in answer_options:
+
+            action_params = option.get('action', {}).get('params')
```
Comment (JamesGardiner) [inline app/validation/validator.py]:
```Python
try:
    action_params = option.get('action', {})['params']
except KeyError:
    continue
```

Comment (JamesGardiner) [reply app/validation/validator.py]: This will keep the code one level less nested

File: app/validation/validator.py [hunk 2]
```diff
@@ -1434,6 +1457,19 @@ def _get_list_names(json_to_validate):
...
         return list_names
 
+    @staticmethod
+    def _get_block_ids(json_to_validate):
+        block_ids = []
+        for section in json_to_validate['sections']:
+            for group in section['groups']:
+                for block in group['blocks']:
+                    block_ids.append(block['id'])
+                    for sub_block in {'add_block', 'edit_block', 'remove_block'}:
```
Comment (ajmaddaford) [inline app/validation/validator.py]: This should also include `add_or_edit_block`

File: app/validation/validator.py (hunk 1 above)
Comment (MebinAbraham) [reply app/validation/validator.py]: I rather keep it consistent as `if`s are already being used to check key existence.
If you really want me to reduce the indentation, I am happy to return early.

``` py
            action_params = option.get('action', {}).get('params')
            if not action_params:
                continue

            list_name = action_params.get('list_name')
```

File: app/validation/validator.py (hunk 2 above)
Comment (MebinAbraham) [reply app/validation/validator.py]: Ah ofc. Added.

File: app/validation/validator.py (hunk 1 above)
Comment (JamesGardiner) [reply app/validation/validator.py]: Yep, good compromise

**Pull request merged**